qemu/tests/create_large_raw_img.py

```diff
@@ -0,0 +1,98 @@
+    :param env: Dictionary with test environment.
+    """
+    def setup_loop_dev(image_path):
+        """
+        Setup a loop device with a file image.
+        :param image_path: The path to the image used to setup loop device
+        :return: The loop device under /dev
+        """
+        cmd_result = process.run("losetup -f %s --show" % image_path)
+        return cmd_result.stdout_text.strip()
+
+    def free_loop_dev(loop_dev):
+        """
+        Free a loop device.
+        :param loop_dev: The loop device will be free
+        """
+        process.run("losetup -d %s" % loop_dev)
+
+    def setup_file_system_env(dev, file_sys, mount_dir):
+        """
+        Setup a new file system environment.
+        :param dev: The device to create new file system
+        :param file_sys: The file system format need to create
+        :param mount_dir: The directory will mount new file system
+        """
+        process.run("mkfs.%s %s" % (file_sys, dev))
+        process.run("mount %s %s" % (dev, mount_dir))
+
+    def clean_env(mount_dir, loop_dev):
+        """
+        Clean the environment.
+        :param mount_dir: The tempory mount directory
+        :param loop_dev: The loop device will be free
+        """
+        process.run("umount %s" % mount_dir)
+        process.run("rm -rf %s" % mount_dir)
+        free_loop_dev(loop_dev)
+
+    def try_create_img(image, file_sys):
+        """
+        Try to create a image file.
+        :param image: The image need to be created
+        :parma err_info: The error info while creating fails
+        """
+        try:
+            image.create(image.params)
+        except exceptions.TestError as err:
+            if params["err_info_on_%s" % file_sys] not in str(err):
+                test.fail("CML failed with unexpected output: %s" % err)
+        else:
+            test.fail(
+                "There is no error when creating an image with large size.")
+
+    filename_img, loop_img = params["images"].split()
+    root_dir = data_dir.get_data_dir()
+    loop = QemuImg(params.object_params(loop_img), root_dir, loop_img)
+    loop_filename = loop.image_filename
+    xfs, ext4 = params["file_systems"].split()
+
+    root_dir = root_dir + "/tmp"
+    process.run("mkdir %s" % root_dir)
+    filename = QemuImg(params.object_params(filename_img), root_dir,
+                       filename_img)
+    loop.create(loop.params)
+    new_dev = setup_loop_dev(loop_filename)
+
+    logging.info("Test creating an image with large size on XFS.")
+    setup_file_system_env(new_dev, xfs, root_dir)
+    try_create_img(filename, xfs)
+    process.run("umount %s" % root_dir)
+
+    logging.info("Test creating an image with large size on EXT4.")
+    setup_file_system_env(new_dev, ext4, root_dir)
+    params["image_size_filename"] = "16T"
+    filename = QemuImg(params.object_params(filename_img), root_dir,
+                       filename_img)
+    try_create_img(filename, ext4)
+
```

Comment:
if you split the config into variants, code from LINE 91 to 96 could be removed.

Reply:
Updated, thanks